dhall/src/Dhall/Core.hs

```diff
@@ -324,6 +327,19 @@ instance Pretty Import where
             Location -> " as Location"
             Code     -> ""
 
+newtype DhallDouble = DhallDouble Double
+    deriving (Show, Data, NFData, Generic)
+
+instance Eq DhallDouble where
+    n == n' =
+        Serialise.serialise (Binary.encode n) ==  Serialise.serialise (Binary.encode n')
```

Comment:
If we switch to a definition that doesn't rely on `Dhall.Binary`, we could probably get rid of `Dhall/Binary.hs-boot` altogether.